sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -162,12 +130,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession)
     assertNoSpecifiedSchema("jdbc")
     // connectionProperties should override settings in extraOptions.
     val params = extraOptions ++ connectionProperties.asScala
-    val options = new JDBCOptions(url, table, params)
-    val parts: Array[Partition] = predicates.zipWithIndex.map { case (part, i) =>
-      JDBCPartition(part, i) : Partition
-    }
-    val relation = JDBCRelation(parts, options)(sparkSession)
-    sparkSession.baseRelationToDataFrame(relation)
+    Dataset.ofRows(sparkSession, UnresolvedJDBCRelation(url, table, predicates, params))
```

Comment:
shall we reuse `UnresolvedDataSource` with the `source` set as `"jdbc"`?

Reply:
hmm the parameters are different. I'm wondering how this is done in SQL. Seems people can't use arbitrary predicates to build JDBCPartition with SQL API.

Reply:
maybe we should add this feature to SQL first, and then unify it.

Reply:
having this available at the moment can help us unify it in SQL afterwards :) This seems to be the only edge case - seems like a new API since I looked at Spark. Do you want me to leave it as is and just migrate the file based and generic data sources?

Reply:
yea let's leave out the JDBC one for now. Once we have feature parity between SQL and DataFrame, we can revisit this.